packages/web/src/common/store/notifications/fetchNotifications.ts

```diff
@@ -46,15 +51,32 @@ export function* fetchNotifications(config: FetchNotificationsParams) {
   const shouldFetchNotificationFromDiscovery =
     useDiscoveryNotifications &&
     discoveryNotificationsGenesisTimestamp &&
-    discoveryNotificationsGenesisTimestamp > Date.parse(timeOffset)
+    Date.parse(timeOffset) > discoveryNotificationsGenesisTimestamp
 
   if (shouldFetchNotificationFromDiscovery) {
     const timestampParam = Math.trunc(Date.parse(timeOffset) / 1000)
+
+    const isRepostOfRepostEnabled = yield* call(
+      getFeatureEnabled,
+      FeatureFlags.REPOST_OF_REPOST_NOTIFICATIONS
+    )
+    const isSaveOfRepostEnabled = yield* call(
+      getFeatureEnabled,
+      FeatureFlags.SAVE_OF_REPOST_NOTIFICATIONS
+    )
+
+    const validTypes = [
+      isRepostOfRepostEnabled ? 'repost_of_repost' : null,
```

Comment:
nit - prob doesnt have to be done for now, but at some point we should put these notif types like `repost_of_repost` and `save_of_repost` into constants in the client